python/cudf/cudf_pandas_tests/test_disable_per_thread_safety.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.
+
+from concurrent.futures import ThreadPoolExecutor
+from time import sleep
+
+import pandas as pd
+
+from cudf.pandas.fast_slow_proxy import _FastSlowProxyMeta
+from cudf.pandas.module_accelerator import disable_module_accelerator
+
+
+def is_enabled(df: pd.DataFrame):
+    return type(type(df)) is _FastSlowProxyMeta
+
+
+def per_thread_work(_):
+    assert is_enabled(pd.DataFrame())
+
+    with disable_module_accelerator():
+        assert not is_enabled(pd.DataFrame())
+
+        # Do some fake work to allow other threads to potentially modify this one
+        for _ in range(1000):
+            sleep(1e-6)
+
+        assert not is_enabled(pd.DataFrame())
+
+        # Ensure that nesting the context manager works too
+        with disable_module_accelerator():
+            assert not is_enabled(pd.DataFrame())
+            for _ in range(1000):
+                sleep(1e-6)
+
+            assert not is_enabled(pd.DataFrame())
+        assert not is_enabled(pd.DataFrame())
+
+    assert is_enabled(pd.DataFrame())
+
+
+def test_disable_pandas_accelerator_multi_threaded():
```

Comment:
I'm curious if you were able to get this test to fail?

Reply:
I mean without the changes to module accelerator

Reply:
Yup this test failed for me locally. If you look at the commit history you'll see that I wrote the test before the fix. I usually try to do that for cases like this because just reproducing the error is half the battle.

Reply:
Oh sweet! Thanks for doing that.